test/query.test.js

```diff
@@ -19,4 +19,38 @@ describe('Testing query route', () => {
     expect(searchTerm.length).toBeGreaterThan(256);
     expect(res.status).toBe(400);
   });
+
+  test('Testing with empty text', async () => {
+    const res = await request(app).get('/query?filter=post');
+    expect(res.status).toBe(400);
+  });
+
+  test('Testing with empty filter', async () => {
+    const searchTerm = encodeURIComponent('I Love Telescope');
+    const res = await request(app).get(`/query?text=${searchTerm}`);
+    expect(res.status).toBe(400);
+  });
+
+  test('Testing with invalid filter', async () => {
+    const searchTerm = encodeURIComponent('I Love Telescope');
+    const res = await request(app).get(`/query?text=${searchTerm}&filter=test`);
+    expect(res.status).toBe(400);
+  });
+
+  test('Testing with valid post filter', async () => {
+    const searchTerm = encodeURIComponent('I Love Telescope');
+    const res = await request(app).get(`/query?text=${searchTerm}&filter=post`);
+    expect(res.status).toBe(200);
+  });
+
+  test('Testing with valid author filter', async () => {
+    const searchTerm = encodeURIComponent('I Love Telescope');
+    const res = await request(app).get(`/query?text=${searchTerm}&filter=author`);
+    expect(res.status).toBe(200);
+  });
+
+  test('Testing with false values', async () => {
+    const res = await request(app).get(`/query?text=&filter=`);
+    expect(res.status).toBe(400);
+  });
```

Comment:
hey @chawlapalak . This is good. One last thing I want to ask from you. Can you squash all your three commit into one commit with some descriptive comment and force push again? Right now the history of commit has something like fixed fail unit tests. Thank you for your effort.

Reply:
Done. Thanks for the feedback and support. Please review the final commit.